servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -3136,7 +3138,9 @@ void RendererSceneRenderRD::_setup_lights(const PagedArray<RID> &p_lights, const
 		if (needs_shadow && in_shadow_range) {
 			// fill in the shadow information
 
-			light_data.shadow_opacity = light_storage->light_get_param(base, RS::LIGHT_PARAM_SHADOW_OPACITY) * shadow_opacity_fade;
+			light_data.shadow_opacity = (p_using_shadows && light_storage->light_has_shadow(base))
```

Comment:
I would add this check to line 3128 above (where ``needs_shadow`` is set) that way this whole block can be skipped when shadows are disabled and shadow opacity will still get set to 0 at line 3203 below

Reply:
Done.